applications/nrf_desktop/src/hw_interface/click_detector.c

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+
+#define MODULE click_detector
+#include "module_state_event.h"
+
+#include "button_event.h"
+#include "power_event.h"
+#include "click_event.h"
+
+#include "click_detector_def.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CLICK_DETECTOR_LOG_LEVEL);
+
+#define CLICK_CHECK_PERIOD	K_MSEC(100)
+
+#define SHORT_CLICK_MAX		K_MSEC(500)
+#define LONG_CLICK_MIN		K_MSEC(5000)
+
+enum state {
+	STATE_DISABLED,
+	STATE_ACTIVE,
+	STATE_SLEEP,
+};
+
+struct key_state {
+	bool pressed;
+	int click_short_timeout;
+	size_t pressed_time;
```

Comment:
`s32_t pressed_time`
`presset_time = -1` instead of `pressed=false`

Reply:
also `s32_t click_short_timeout`
s32_t is used for time in zephyr

Reply:
ok